### PR TITLE
Post Deployment Initialization function

### PR DESCRIPTION
**Pull request by axthosarouris**
This is the very first version of a stack initialization function. The function belongs to the stack because it will be easier to use the stack resources. 

### Background:
In order for the initialization function to be executed after a successful pipeline, there are two alternatives:
1. The pipeline calls the function directly
     This method requires that we know from before the name (and therefore the ARN) of both the stack and the init function before the stack is created. In other words, each stack should have deterministic name and each init function should have a deterministic name as well.  
2. The function listens to pipeline events and is activated when a pipeline execution is successful.
   This approach requires that the pipeline name is passed as a parameter to the stack. Probably this is an easier way to go. 

To achieve the second approach we use CodePipeline EventBridge events as they are described here: https://docs.aws.amazon.com/codepipeline/latest/userguide/detect-state-changes-cloudwatch-events.html

### This PR.

This PR contains only the logic of a very simple handler, that logs the event. The connection to the pipeline is not yet implemented but it will follow.

### CODE REVIEW

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/initialization/InitializationHandler.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package no.unit.nva.publication.events.handlers.initialization;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import no.unit.nva.events.handlers.EventHandler;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InitializationHandler extends EventHandler<PipelineEvent, Void> {
```
Comment (axthosarouris) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/initialization/InitializationHandler.java:9]: Probably we can use the event logic we already have

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/initialization/PipelineEvent.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package no.unit.nva.publication.events.handlers.initialization;
+
+public class PipelineEvent {
+
+    private String pipeline;
```
Comment (axthosarouris) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/initialization/PipelineEvent.java:5]: Useful information of the pipeline event.

File: publication-event-handlers/src/test/resources/initialization/pipeline_succeeded_event.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (axthosarouris) [inline publication-event-handlers/src/test/resources/initialization/pipeline_succeeded_event.json:1]: Sample event copied from https://docs.aws.amazon.com/codepipeline/latest/userguide/detect-state-changes-cloudwatch-events.html

File: template.yaml [hunk 4]
```diff
@@ -399,11 +399,11 @@ Resources:
       Handler: no.unit.nva.publication.fetch.FetchPublicationHandler::handleRequest
       Runtime: java11
       MemorySize: 1408
-#      AutoPublishAlias: live
-#      DeploymentPreference:
-#        Type: AllAtOnce
-#      ProvisionedConcurrencyConfig:
-#        ProvisionedConcurrentExecutions: 1
+      AutoPublishAlias: live
```
Comment (axthosarouris) [inline template.yaml:402]: Part of another PR that is not yet merged

**Review by brinxmat — APPROVED** [review]

**Pull request merged**